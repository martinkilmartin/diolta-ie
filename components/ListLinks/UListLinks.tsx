import { Link } from '../Link'

type Props = {
  links?: string[]
}

const UListLinks = ({ links = [] }: Props): JSX.Element => (
  <ul className="flex flex-wrap justify-between max-w-screen-md mx-auto text-lg font-light">
    {links.map((link, index) => (
      <Link text={link} key={`${link.substring(0, 4)}_${index}`} />
    ))}
  </ul>
)

export default UListLinks
