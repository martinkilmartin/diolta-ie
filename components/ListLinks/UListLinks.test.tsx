import renderer from 'react-test-renderer'
import UListLinks from './UListLinks'

test('UListLinks exists with class', () => {
  const uListLinksText = [
    'ABCDEF',
    'HIJKLMNOP',
    'QRS',
    'TUVWXYZ',
    '01234',
    '56789',
  ]
  const uListLinksRenderer = renderer.create(
    <UListLinks links={uListLinksText} />
  )
  const tree = uListLinksRenderer.toJSON()
  expect(tree).toMatchSnapshot()
  const uListLinksInstance = uListLinksRenderer.root
  expect(
    uListLinksInstance.findByProps({
      className:
        'flex flex-wrap justify-between max-w-screen-md mx-auto text-lg font-light',
    }).children.length
  ).toEqual(6)
})
