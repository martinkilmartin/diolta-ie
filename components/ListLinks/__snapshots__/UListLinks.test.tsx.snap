// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UListLinks exists with class 1`] = `
<ul
  className="flex flex-wrap justify-between max-w-screen-md mx-auto text-lg font-light"
>
  <a
    className="transition-colors duration-200 hover:text-accent-focus"
    href="#"
  >
    ABCDEF
  </a>
  <a
    className="transition-colors duration-200 hover:text-accent-focus"
    href="#"
  >
    HIJKLMNOP
  </a>
  <a
    className="transition-colors duration-200 hover:text-accent-focus"
    href="#"
  >
    QRS
  </a>
  <a
    className="transition-colors duration-200 hover:text-accent-focus"
    href="#"
  >
    TUVWXYZ
  </a>
  <a
    className="transition-colors duration-200 hover:text-accent-focus"
    href="#"
  >
    01234
  </a>
  <a
    className="transition-colors duration-200 hover:text-accent-focus"
    href="#"
  >
    56789
  </a>
</ul>
`;
